### PR TITLE
Standarizing reserve names; putting in (mostly) a single place

### PR DESCRIPTION
**Pull request by bknueven**
## Summary/Motivation:
After #263 I realized the list of ancillary services was really in two place. This standardizes it to `egret/model_library/transmission/tx_utils.py`, which already knows a lot about the underlying model data structure. Along the way I noticed that `supplemental_reserve` breaks the `*_requirement`/`*_price`/`*_shortfall` pattern, so this changes the name of `supplemental_price` and `supplemental_shortfall` to match that pattern.

## Changes proposed in this PR:
- Centralize list of ancillary services to tx_utils
- Rename `supplemental_price` to `supplemental_reserve_price` and `supplemental_shortfall` to `supplemental_reserve_shortfall` (including json files).

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://github.com/grid-parity-exchange/Egret/blob/main/CONTRIBUTING.md) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by darrylmelander — CHANGES_REQUESTED** [review]
Up to you whether to remove the ancillary_service_list. I only marked this review as "Request Changes" in case you had it set to auto-merge before you had a chance to decide.

File: egret/model_library/transmission/tx_utils.py [hunk 1]
```diff
@@ -214,37 +214,38 @@ def load_shed_limit(load, gens, gen_mins):
...
+penalty_prices = (
+    'regulation_penalty_price',
+    'spinning_reserve_penalty_price',
+    'non_spinning_reserve_penalty_price',
+    'supplemental_reserve_penalty_price',
+    'flexible_ramp_penalty_price',
+)
+
+# for ../unit_commitment/services.py -- tuple for safety, name for compatability
+ancillary_service_list = ( name+'_requirement' for name in ancillary_services[1:] )
```
Comment (bknueven) [inline egret/model_library/transmission/tx_utils.py:239]:
```suggestion
```

File: egret/model_library/unit_commitment/services.py [hunk 2]
```diff
@@ -11,20 +11,10 @@
 from pyomo.environ import *
 import math
 
+from egret.model_library.transmission.tx_utils import ancillary_service_list
```
Comment (bknueven) [inline egret/model_library/unit_commitment/services.py:14]:
```suggestion
```

**Review by bknueven — COMMENTED** [review]
Thanks @darrylmelander. I deleted `ancillary_service_list` as you suggested.

**Review by darrylmelander — APPROVED** [review]

**Pull request merged**